### PR TITLE
Correctly animate title's movement in about screen's collapsing header

### PR DESCRIPTION
**Pull request by saket**
It'll be easier to review the diff in split mode. 

https://github.com/ZacSweers/CatchUp/assets/2387680/17246c6e-f0a0-43a6-b0d7-60f2d0ca2359

### CODE REVIEW

File: app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt [hunk 1]
```diff
@@ -64,353 +42,143 @@ import androidx.compose.ui.text.UrlAnnotation
...
-        onDragStopped = { velocity ->
-          settleAppBar(
-            scrollBehavior.state,
-            velocity,
-            scrollBehavior.flingAnimationSpec,
-            scrollBehavior.snapAnimationSpec
+  val scaffoldState = rememberCollapsingToolbarScaffoldState()
+  val parallaxRatio = 0.2f
+
+  Surface(modifier) { // todo: why is this needed when AboutScreen is already using Scaffold()?
```
Comment (saket) [inline app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt:71]: I didn't get a chance to investigate this further

File: app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt [hunk 2]
```diff
@@ -64,353 +42,143 @@ import androidx.compose.ui.text.UrlAnnotation
...
-                    IntOffset(
-                      x = coordinates.positionInParent().x.roundToInt(),
-                      y = coordinates.positionInParent().y.roundToInt()
-                    )
+          Text(
+            modifier =
+              Modifier.align(CenterHorizontally)
+                .onGloballyPositioned { titleCoordinates = it.positionInParent() }
+                .offset {
+                  lerp(IntOffset.Zero, textOffset, 1f - scaffoldState.toolbarState.progress)
```
Comment (saket) [inline app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt:138]: I tried using `arcLerp` here, but it's producing a terrible result.

File: app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt (hunk 1 above)
Comment (ZacSweers) [reply app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt:71]: What happens without it?

File: app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt (hunk 2 above)
Comment (ZacSweers) [reply app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt:138]: it's extremely scrapped together so not surprising, I'm not sure I did it right

**Review by ZacSweers — APPROVED** [review]
This looks so good, thanks a ton!

File: app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt (hunk 1 above)
Comment (saket) [reply app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt:71]: IIRC the header's background color and its content colors were black on black

**Pull request merged**